pkg/util/testutil/timeout.go

```diff
@@ -13,6 +13,18 @@ import (
 // Condition is evaluated in a goroutine to avoid tests hanging if a system
 // is deadlocked.
 func AssertTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	internalTrueBeforeTimeout(t, false, frequency, timeout, condition)
+}
+
+// RequireTrueBeforeTimeout is the same as AssertTrueBeforeTimeout, but it calls
+// t.failNow() if the condition function times out.
+func RequireTrueBeforeTimeout(t *testing.T, frequency, timeout time.Duration, condition func() bool) {
+	if !internalTrueBeforeTimeout(t, true, frequency, timeout, condition) {
+		t.FailNow()
+	}
+}
+
+func internalTrueBeforeTimeout(t *testing.T, require bool, frequency, timeout time.Duration, condition func() bool) bool {
 	t.Helper()
```

Comment:
you _may_ have to call `t.Helper()` from `AssertTrueBeforeTimeout` and `RequireTrueBeforeTimeout` as well, so that the functions are skipped when `go test` prints file and function information. You'll want to check what's printed when the test fails.